database/helpers.go

```diff
@@ -0,0 +1,81 @@
+package database
```

Comment:
"helpers" is very general name for this file, while the contents are pretty specific.

I'd call it "keys.go" or something. 
Preferably find a name that shows that it's both keys and buckets, but I can't think of a good one, and since buckets are just something to help build keys, I guess it's fine.